owncloudData/src/test/java/com/owncloud/android/data/spaces/datasource/implementation/OCLocalSpacesDataSourceTest.kt

```diff
@@ -0,0 +1,435 @@
+    )
+
+    @Before
+    fun init() {
+        val db = mockkClass(OwncloudDatabase::class)
+
+        every { db.spacesDao() } returns spacesDao
+
+        ocLocalSpacesDataSource = OCLocalSpacesDataSource(spacesDao)
+    }
+
+    @Test
+    fun `saveSpacesForAccount inserts spaces and special spaces returns unit`() {
+        val spaceEntities = mutableListOf<SpacesEntity>()
+        val spaceSpecialEntities = mutableListOf<SpaceSpecialEntity>()
+
+        listSpaceSpecial.forEach { spaceModel ->
+            spaceEntities.add(spaceModel.toEntity())
+            spaceModel.special?.let { listOfSpacesSpecials ->
+                spaceSpecialEntities.addAll(listOfSpacesSpecials.map { it.toEntity(spaceModel.accountName, spaceModel.id) })
+            }
+        }
+
+        every {
+            spacesDao.insertOrDeleteSpaces(any(), any())
+        } just Runs
+
+        ocLocalSpacesDataSource.saveSpacesForAccount(listSpaceSpecial)
+
+        verify(exactly = 1) {
+            spacesDao.insertOrDeleteSpaces(spaceEntities, spaceSpecialEntities)
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `saveSpacesForAccount inserts spaces and special spaces returns an exception when dao receive an exception`() {
+        val spaceEntities = mutableListOf<SpacesEntity>()
+        val spaceSpecialEntities = mutableListOf<SpaceSpecialEntity>()
+
+        listSpaceSpecial.forEach { spaceModel ->
+            spaceEntities.add(spaceModel.toEntity())
+            spaceModel.special?.let { listOfSpacesSpecials ->
+                spaceSpecialEntities.addAll(listOfSpacesSpecials.map { it.toEntity(spaceModel.accountName, spaceModel.id) })
+            }
+        }
+
+        every {
+            spacesDao.insertOrDeleteSpaces(any(), any())
+        } throws Exception()
+
+        ocLocalSpacesDataSource.saveSpacesForAccount(listSpaceSpecial)
+    }
+
+    @Test
+    fun `getPersonalSpaceForAccount by drive type returns a OCSpace`() {
+        every {
+            spacesDao.getSpacesByDriveTypeForAccount(any(), any())
+        } returns listOf(OC_SPACE_PERSONAL.toEntity())
+
+        val resultActual = ocLocalSpacesDataSource.getPersonalSpaceForAccount(OC_ACCOUNT_NAME)
+
+        assertEquals(OC_SPACE_PERSONAL, resultActual)
+
+        verify(exactly = 1) {
+            spacesDao.getSpacesByDriveTypeForAccount(OC_ACCOUNT_NAME, setOf(OCSpace.DRIVE_TYPE_PERSONAL))
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getPersonalSpaceForAccount by drive type returns an exception when dao receive an exception`() {
+        every {
+            spacesDao.getSpacesByDriveTypeForAccount(any(), any())
+        } throws Exception()
+
+        ocLocalSpacesDataSource.getPersonalSpaceForAccount(OC_ACCOUNT_NAME)
+    }
+
+
+    @Test
+    fun `getSharesSpaceForAccount returns a OCSpace`() {
+        every {
+            spacesDao.getSpaceByIdForAccount(SPACE_ID_SHARES, OC_ACCOUNT_NAME)
+        } returns OC_SPACE_PERSONAL.toEntity()
```

Comment:
Well, this test is passing but in concept is not correct. We're trying to retrieve the shares space, not the personal space. Use an example of shares space instead

Reply:
done

Reply:
Done